src/main/java/org/apache/jmeter/protocol/aws/kinesis/KinesisRecordProducerSampler.java

```diff
@@ -0,0 +1,186 @@
+                });
+
+        log.info("Create Kinesis Producer.");
+        kinesisClient = (KinesisClient) createSdkClient(credentials);
+    }
+
+    /**
+     * Main method to execute the test on single thread. Create Data Records and publish it in Kinesis stream.
+     * @param context
+     *        Arguments values on Java Sampler.
+     * @return SampleResult, captures data such as whether the test was successful,
+     * the response code and message, any request or response data, and the test start/end times
+     */
+    @Override
+    public SampleResult runTest(JavaSamplerContext context) {
+
+        SampleResult result = newSampleResult();
+        sampleResultStart(result, String.format("Stream Name: %s \nPartition Key: %s \nData Record: %s",
+                context.getParameter(KINESIS_STREAM_NAME),
+                context.getParameter(KINESIS_PARTITION_KEY),
+                context.getParameter(KINESIS_DATA_RECORD)));
+
+        try {
+            log.info("Publishing Data Records.");
+            PutRecordsResponse response = kinesisClient.putRecords(createPutRecordRequest(context));
+            sampleResultSuccess(result,String.format("Response details: %s \nEncryption Type: %s",
+                    response.toString(),
+                    response.encryptionTypeAsString()));
+        }catch (KinesisException e){
+            sampleResultFail(result, e.awsErrorDetails().errorCode(), e.awsErrorDetails().errorMessage());
+        }
+
+        return result;
+    }
+
+    /**
+     * Close AWS Kinesis Data Stream Client after run single thread.
+     * @param context
+     *        Arguments values on Java Sampler.
+     */
+    @Override
+    public void teardownTest(JavaSamplerContext context) {
+        log.info("Close Kinesis Producer.");
+        kinesisClient.close();
+    }
+
+    /**
+     * Create PutRecordsRequest with stream name, partition key and data.
+     * @param context
+     *        Arguments values on Java Sampler.
+     * @return PutRecordsRequest
+     */
+    public PutRecordsRequest createPutRecordRequest(JavaSamplerContext context) {
+        String tempPayload = context.getParameter(KINESIS_DATA_RECORD);
+        JSONArray jsonArray = new JSONArray(tempPayload);
+        List<PutRecordsRequestEntry> putRecordsRequestEntryList = new ArrayList<>();
+        PutRecordsRequest putRecordsRequest = null;
+        for (int i = 0; i < jsonArray.length(); i++) {
+            try {
+                putRecordsRequestEntryList.add(PutRecordsRequestEntry.builder()
+                        .data(SdkBytes.fromByteArray(String.valueOf(jsonArray.get(i)).getBytes(StandardCharsets.UTF_8)))
+                        .partitionKey(context.getParameter(KINESIS_PARTITION_KEY) + "-" + i).build());
+
+
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+            putRecordsRequest = PutRecordsRequest.builder()
```

Comment:
Looks like you are building PutRecordRequest for each record in the array.
I guess is better to create PutRecordRequest once with the List<PutRecordsRequestEntry and streamName.